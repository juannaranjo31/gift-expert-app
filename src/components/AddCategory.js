import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if (inputValue.trim().length > 2) {
            setCategories( c => [inputValue, ...c]);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <input 
            type="text"
            value={inputValue}
            onChange={ (e) => setInputValue( e.target.value ) }
            placeholder='Ingrese una categoria para su busqueda de gifts :)'
            
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
