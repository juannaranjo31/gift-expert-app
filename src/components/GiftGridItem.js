import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ( {title, url} ) => {

    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={url} alt={title}></img>
            <p>{title}</p>
        </div>
    )
}

GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}