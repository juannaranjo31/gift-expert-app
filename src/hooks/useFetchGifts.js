import { useState, useEffect } from 'react';
import { getGifts } from '../helpers/getGifts';

export const useFetchGifts = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {

        setTimeout(() => {
            getGifts(category)
                .then(imgs => setstate({
                    data: imgs,
                    loading: false
                }))
        }, 500)

    }, [category]);


    return state;
}