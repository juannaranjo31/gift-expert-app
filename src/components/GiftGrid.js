import React from 'react'
import PropTypes from 'prop-types';
import { useFetchGifts } from '../hooks/useFetchGifts';
import { GiftGridItem } from './GiftGridItem';

export const GiftGrid = ({ category }) => {

    const {data:images, loading} = useFetchGifts( category );

    return (
        <>
            <h3> {category} </h3>
            {loading && <p className='animate__animated animate__flash'>Loading...</p>}

            <div className='card-grid'>
                {
                    images.map((img) => {
                        return <GiftGridItem key={img.id} {...img} />
                    })
                }
            </div>

        </>
    )
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
};