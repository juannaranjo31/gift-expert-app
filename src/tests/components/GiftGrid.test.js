import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from "enzyme";
import { useFetchGifts } from '../../hooks/useFetchGifts';
import { GiftGrid } from "../../components/GiftGrid";
jest.mock('../../hooks/useFetchGifts');

describe('Pruebas en <GiftGrid />', () => {
    const category = 'Dragon ball';

    test('Debe reenderizarse correctamente', () => {
        useFetchGifts.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GiftGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imagenes con useFetchGifts', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/algo.jpg',
                title: 'cualquier cosa'
            },

            {
                id: '123',
                url: 'https://localhost/algo.jpg',
                title: 'cualquier cosa'
            }

        ];

        useFetchGifts.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GiftGrid category={category} />);
        
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GiftGridItem').length).toBe(gifs.length);

    });


});
