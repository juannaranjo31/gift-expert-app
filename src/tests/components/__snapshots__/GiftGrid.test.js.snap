// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGrid /> Debe reenderizarse correctamente 1`] = `
<Fragment>
  <h3>
     
    Dragon ball
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
